### ISSUE TITLE
Unexpected incompatibility of identical `TypedDict`s

### ISSUE THREAD

**Issue by anniel-stripe** [microsoft/pyright#6416]
**Describe the bug**
Hello! We discovered an unexpected Pyright error related to `TypedDict`s:

**Code or Screenshots**
Minimal repro ([Pyright playground](https://pyright-play.net/?code=GYJw9gtgBALgngBwJYDsDmUkQWEMoAqiApgCYAiSAxjADRQCqKSYK9AMkjMSAIYA29APIIYLFAIBQ0qv14BneVABiYMAEYAFEQRlKNAJQAuSVDNQARkcwoY0ybIVLVYAEzaSFajGOnzVmzsZOUUofRgtHT1vX3MoXmt5GBA-M2A1axExVgEAbRd1AF17R1Dw9yivQxM4hKgklLj0sEzRcTyXV2LpUmJgKAALYn5%2BME1Sa3LY8wQQVBhx3IAiXiXCg3sJsO91KABeKABvVLioFaXrJaGRsCXaE7il5oujh9OzJYsX9QAGN7MAL4nIGSa6jcYbIA)):
```python
from typing import TypedDict, Union, Literal


class Foo1(TypedDict):
    b: int


class Foo2(TypedDict):
    b: int


class Dict1(TypedDict):
    a: str
    foo: Union[Literal[''], Foo1]


class Dict2(TypedDict):
    a: str
    foo: Union[Literal[''], Foo2]


def hello(d: Dict2):
    print(d["a"])


d: Dict1 = {
        "a": "hello",
        "foo": {
            "b": 10
        }
    }

hello(d)
```

produces this error:
```python
Argument of type "Dict1" cannot be assigned to parameter "d" of type "Dict2" in function "hello"
  "foo" is an incompatible type  (reportGeneralTypeIssues)
```

**VS Code extension or command-line**
Are you running pyright as a VS Code extension, a language server in another editor, integrated into Pylance, or the command-line tool? Which version?
command-line tool
pyright 1.1.334

**Comment by erictraut** [comment]
Thanks for the bug report. I'm able to repro the problem, and I agree it's a bug. I'll investigate further.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.337, which I just published. It will also be included in a future release of pylance.
